tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -918,11 +924,66 @@ public function it_should_create_mime_types_for_allowed_sizes_only_via_global_va
 		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
 	}
 
+	/**
+	 * Test image quality for image conversion.
+	 *
+	 * @test
+	 */
+	public function it_should_set_quality_with_image_conversion() {
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png', array( 'mime_type' => 'image/png' ) );
+
+		// Quality setting for the source image. For PNG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for PNG is 82.' );
+
+		// Temporary file.
+		$file = wp_tempnam();
```

Comment:
Why do we need to create this file here? I would suggest we simply pass a file name or potentially `null` to `WP_Image_Editor::save()`.

Creating a temporary file just for this seems unnecessary.

Reply:
When we save the editor, it generates the image for WebP for that mime and it is generated in our testdata folder.
<img width="283" alt="Screenshot 2022-11-15 at 4 20 38 PM" src="https://user-images.githubusercontent.com/4090336/201902032-acd9365a-6815-4324-aec5-8c19cbf77524.png">

so to avoid generating these files here better to generate them in temp folder and even if unlink code doesn't work if assertions fails, temp folder will take of itself, I guess.

Reply:
Also don't understand why this is needed here

Reply:
I talked about this with @aaemnnosttv, makes sense to me now. The point is that this function generates a unique filename in a temporary directory. If we don't do this, the files are generated within our repository directory, which is not advisable as then they may by accident end up in our repository.

WordPress core uses a similar mechanism for some of its tests, then also unlinking the file, so that seems good to me now. 👍